backend/routes/appointments/viewAppointmentsRouter.js

```diff
@@ -51,10 +51,52 @@ module.exports = ({ prisonApi, whereaboutsApi, logError }) => async (req, res) =
       court: appointment.court,
     }))
 
+  const addBookingIdAndHearingTypesToVLBAppointments = appointment => {
+    const appCopy = { ...appointment }
+    if (appointment.appointmentTypeCode === 'VLB') {
+      videoLinkAppointments.forEach(vlAppointment => {
+        if (vlAppointment.appointmentId === appointment.id) {
+          appCopy.hearingType = vlAppointment.hearingType
+          appCopy.bookingId = vlAppointment.bookingId
```

Comment:
Probably best to avoid reference to a copy - you could just return the existing structure with extra data (as i show in the comment below) right here. You would also change the return statement at the bottom to use the "appointment" variable.